src/order/bounded_lattice.lean

```diff
@@ -182,31 +181,38 @@ lemma ne.bot_lt' (h : ⊥ ≠ a) : ⊥ < a := h.symm.bot_lt
 
 end order_bot
 
-lemma strict_mono.minimal_preimage_bot [linear_order α] [order_bot β]
+lemma strict_mono.minimal_preimage_bot [linear_order α] [partial_order β] [order_bot β]
   {f : α → β} (H : strict_mono f) {a} (h_bot : f a = ⊥) (x : α) :
   a ≤ x :=
 H.minimal_of_minimal_image (λ p, by { rw h_bot, exact bot_le }) x
 
-theorem order_bot.ext_bot {α} {A B : order_bot α}
+theorem order_bot.ext_bot {α} [partial_order α] (A B : order_bot α)
   (H : ∀ x y : α, (by haveI := A; exact x ≤ y) ↔ x ≤ y) :
   (by haveI := A; exact ⊥ : α) = ⊥ :=
 bot_unique $ by rw ← H; apply bot_le
 
-theorem order_bot.ext {α} {A B : order_bot α}
+theorem order_bot.ext {α} [partial_order α] {A B : order_bot α}
   (H : ∀ x y : α, (by haveI := A; exact x ≤ y) ↔ x ≤ y) : A = B :=
 begin
   have := partial_order.ext H,
-  have tt := order_bot.ext_bot H,
-  casesI A, casesI B,
-  injection this; congr'
+  have tt := order_bot.ext_bot A B H,
+  casesI A with a ha, casesI B with b hb,
+  congr,
+  exact le_antisymm (ha _) (hb _)
 end
 
 /-- A `semilattice_sup_top` is a semilattice with top and join. -/
-class semilattice_sup_top (α : Type u) extends order_top α, semilattice_sup α
+class semilattice_sup_top (α : Type u) extends semilattice_sup α, has_top α :=
+(le_top : ∀ a : α, a ≤ ⊤)
 
 section semilattice_sup_top
 variables [semilattice_sup_top α] {a : α}
 
+@[priority 100] -- see Note [lower instance priority]
+instance semilattice_sup_top.to_order_top : order_top α :=
```

Comment:
Why not extending? Doesn't matter much as you'll be refactoring that next, I guess.

Reply:
Same extending issue that I mentioned above that `extending` doesn't work with `order_top` or `order_bot`.

Reply:
Ah yeah sorry, I'd forgotten.